components/ws-manager-mk2/controllers/workspace_controller.go

```diff
@@ -102,6 +102,14 @@ func (r *WorkspaceReconciler) Reconcile(ctx context.Context, req ctrl.Request) (
 		workspace.Status.Conditions = []metav1.Condition{}
 	}
 
+	if r.OnReconcile != nil {
+		// Publish to subscribers in a goroutine, to prevent blocking the main reconcile loop.
+		ws := workspace.DeepCopy()
+		go func() {
+			r.OnReconcile(ctx, ws)
+		}()
```

Comment:
also moved sending the updates to a separate goroutine, so that any slow receivers don't block the reconciler